GameServerCore/Domain/GameObjects/ITarget.cs

```diff
@@ -10,5 +10,6 @@ public interface ITarget
 
         float GetDistanceTo(ITarget target);
         Vector2 GetPosition();
+        bool WithinRange(Vector2 from, Vector2 to, float range);
```

Comment:
If is this a static method why is it in ITarget? or just use ITarget position.

Reply:
It is used for abilities that can be cast at any range but that shouldn’t be, like Shaco W.

Reply:
still, this function is sort of helper function. Either use some variables from ITarget/classes implement this function, or make this function static.

Reply:
maybe ITarget can just inherit Vector2 no?

Reply:
As far as I'm aware, an interface can't inherit from a class. I had the same idea starting out but scrapped it due to that. You could however make an IVector2 interface and use that if you wanted to, then inherit from that in both Vector2 and ITarget.

I'm also most likely for the extension function approach, alternatively a method on Vector2 and an extension function that takes and converts ITarget to Vector2 then passes that to the method on Vector2, that way keeping Vector2 cleaner.

Reply:
I use WithinRange for if spell.X and spell.Y is within correct cast range.

Reply:
The point is that this is `static` function when you want to use non-static in ITarget. That creates some ambiguity so we need this as extended or helper function. (some changes in hierarchy needed)